lib/taskjuggler/TjpSyntaxRules.rb

```diff
@@ -3267,6 +3267,23 @@ def rule_projectBodyAttributes
     arg(1, 'date', 'Alternative date to be used as current date for all ' +
         'computations')
 
+    pattern(%w( _ref !date ), lambda {
+      @project['ref'] = @val[1]
+      @scanner.addMacro(TextParser::Macro.new('ref', @val[1].to_s,
+                                              @sourceFileInfo[0]))
+      @scanner.addMacro(TextParser::Macro.new(
+        'today', @val[1].to_s(@project['timeFormat']), @sourceFileInfo[0]))
+    })
+    doc('ref', <<'EOT'
+Specify the reference date that TaskJuggler uses for calculation as date that can be
```

Comment:
I'm not sure I understand this. These patches merely cause a line to be drawn in the Gantt charts. It does not affect any calculations.